packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/Palette.tsx

```diff
@@ -126,7 +130,19 @@ export const Palette = ({
     x = (paletteX - viewportX) / viewportZoom;
     y = (paletteY - viewportY) / viewportZoom;
   }
-  const { handleToolClick, palette } = usePalette({ x, y, diagramElementId, onDirectEditClick, targetObjectId });
+
+  const { palette }: UsePaletteContentValue = usePaletteContents(diagramElementId);
+  const { setLastToolInvoked } = useDiagramPalette();
```

Comment:
It would be nice to get rid or this last reference to the diagram palette from the generic palette component but we are moving in the right direction :+1: